src/tests/commonTests.ts

```diff
@@ -531,14 +536,38 @@ export async function formAssociated(componentTagOrHtml: TagOrHTML, options: For
     typeof options.testValue === "boolean" && (await page.$eval(componentTag, (component) => "checked" in component));
   const resettablePropName = checkable ? "checked" : "value";
   const initialValue = await component.getProperty(resettablePropName);
+  const name = await component.getProperty("name");
 
+  await assertValueSubmissionType();
   await assertValueResetOnFormReset();
   await assertValueSubmittedOnFormSubmit();
 
   if (options.submitsOnEnter) {
     await assertFormSubmitOnEnter();
   }
 
+  async function assertValueSubmissionType(): Promise<void> {
+    const inputType = options.inputType ?? "text";
+
+    const hiddenFormInputType = await page.evaluate(
+      async (inputName: string, hiddenFormInputSlotName: string): Promise<string> => {
+        const hiddenFormInput = document.querySelector<HTMLInputElement>(
+          `[name="${inputName}"] input[slot=${hiddenFormInputSlotName}]`
+        );
+
+        return hiddenFormInput.type;
+      },
+      name,
+      hiddenFormInputSlotName
+    );
+
+    if (checkable) {
+      expect(hiddenFormInputType).toMatch(/radio|checkbox/);
```

Comment:
Would `expect(["radio", "checkbox"]).toContain(hiddenFormInputType)` work here so we don't need to use regex?

Reply:
I would definitely change it if the regex was slightly more complex. For now, both seem pretty straightforward. We can revisit this later.